ui/views/tx/submit/horizon-submit-tx-view.js

```diff
@@ -5,20 +5,18 @@ import {checkTxSubmitted} from '../../../infrastructure/tx-dispatcher'
 import config from '../../../app.config.json'
 import {horizonErrorHandler} from './horizon-error-handler'
 
-export default withErrorBoundary(function HorizonSubmitTxView({txInfo}) {
+export default withErrorBoundary(function HorizonSubmitTxView({txInfo, onUpdate}) {
     const {readyToSubmit, hash, submit, submitted, xdr, status, network, error} = txInfo
     const [inProgress, setInProgress] = useState(false)
     const [isExist, setIsExist] = useState(true)
 
     useEffect(() => {
-        if (!txInfo.submit && !txInfo.submitted) {
+        if (!submit && !submitted) {
             //check existence of transaction in Horizon
             checkTxSubmitted(txInfo)
-                .then(tx => {
-                    setIsExist(!!tx.submitted)
-                })
+                .then(tx => setIsExist(!!tx.submitted))
         }
-    }, [txInfo])
+    }, [submit, submitted])
```

Comment:
```suggestion
    }, [hash, status, submit, submitted])
```
Otherwise, might not work properly with navigation